src/preferences/PreferencesBase.js

```diff
@@ -189,7 +203,13 @@ define(function (require, exports, module) {
                         try {
                             result.resolve(JSON.parse(text));
                         } catch (e) {
-                            result.reject(new ParsingError("Invalid JSON settings at " + path + "(" + e.toString() + ")"));
+                            if (isStateJson && createIfNew) {
```

Comment:
We check this flag before calling 'result.resolve({})' which eventually (after multiple calls) saves the new 'state.json' file.
I agree the flag name is not very clear. Renaming it to a more appropriate 'createIfMissing'.